pkg/controller/job/job_controller_test.go

```diff
@@ -3045,13 +3095,138 @@ func TestSyncJobWithJobPodFailurePolicy(t *testing.T) {
 					},
 				},
 			},
-			wantStatusActive: 1, // This is a replacement Pod: the terminating Pod is neither active nor failed.
+			wantStatusActive:      1, // This is a replacement Pod: the terminating Pod is neither active nor failed.
+			wantStatusTerminating: pointer.Int32(1),
+		},
+		"default value for podreplacementpolicy is Failed when not specifying PodReplacementPolicy": {
+			enableJobPodFailurePolicy:     true,
+			enablePodDisruptionConditions: true,
+			enableJobPodReplacementPolicy: true,
+			job: batch.Job{
+				TypeMeta:   metav1.TypeMeta{Kind: "Job"},
+				ObjectMeta: validObjectMeta,
+				Spec: batch.JobSpec{
+					Parallelism:  pointer.Int32(1),
+					Selector:     validSelector,
+					Template:     validTemplate,
+					BackoffLimit: pointer.Int32(0),
+					PodFailurePolicy: &batch.PodFailurePolicy{
+						Rules: []batch.PodFailurePolicyRule{
+							{
+								Action: batch.PodFailurePolicyActionCount,
+								OnPodConditions: []batch.PodFailurePolicyOnPodConditionsPattern{
+									{
+										Type:   v1.DisruptionTarget,
+										Status: v1.ConditionTrue,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			pods: []v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						DeletionTimestamp: &now,
+					},
+					Status: v1.PodStatus{
+						Phase: v1.PodRunning,
+					},
+				},
+			},
+			wantStatusTerminating: pointer.Int32(1),
+		},
+		"count terminating pods even if terminating Pod considered failed when PodDisruptionConditions is disabled": {
+			enableJobPodFailurePolicy:     true,
+			enableJobPodReplacementPolicy: true,
+			job: batch.Job{
+				TypeMeta:   metav1.TypeMeta{Kind: "Job"},
+				ObjectMeta: validObjectMeta,
+				Spec: batch.JobSpec{
+					Parallelism:          pointer.Int32(1),
+					Selector:             validSelector,
+					Template:             validTemplate,
+					BackoffLimit:         pointer.Int32(0),
+					PodReplacementPolicy: (*batch.PodReplacementPolicy)(pointer.String("Failed")),
+					PodFailurePolicy: &batch.PodFailurePolicy{
+						Rules: []batch.PodFailurePolicyRule{
+							{
+								Action: batch.PodFailurePolicyActionCount,
+								OnPodConditions: []batch.PodFailurePolicyOnPodConditionsPattern{
+									{
+										Type:   v1.DisruptionTarget,
+										Status: v1.ConditionTrue,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			pods: []v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						DeletionTimestamp: &now,
+					},
+					Status: v1.PodStatus{
+						Phase: v1.PodRunning,
+					},
+				},
+			},
+			wantStatusTerminating: pointer.Int32(1),
+		},
+		"count terminating pods even if terminating Pod not considered failed when PodDisruptionConditions is enabled": {
```

Comment:
This test case looks duplicated with the second one above, am I missing something?

Reply:
The only difference is `PodDisruptionCondition` being flipped from false to true.

Reply:
I mean the two test cases up :), the "default value for podreplacementpolicy is Failed when not specifying PodReplacementPolicy" and "count terminating pods even if terminating Pod not considered failed when PodDisruptionConditions is enabled"

Reply:
Good call.  It was a duplicate.

Reply:
Pushed up a change to remove.